File: runme/mcgen/local/datan/exasrc/e6gr.c
#include <stdio.h>
#include <math.h>
#include <string.h>
#include "grsrc.h"
#include "datsrc.h"

/* Table of constant values */

static integer c__1 = 1, c__3 = 3, c__9 = 9;

main()
{
    /* System generated locals */
    integer i__1, i__2;

    /* Local variables */
    static integer i, lcapt, nx, npl, nws, ltx, lty;
    static double delx, hist[100], sigma, x0, alambd, dx, xpl[1000], ypl[1000];
    static char capt[75], tx[75], ty[75];

/* identify program to user */
    (void)printf("%s\n\n", "Program E6GR demonstrates use of GRHSCV.");
/* ask for number of workstation */
    grnbws_();
    (void)printf("%s\n", "Please, enter number of workstation:");
    (void)printf("%s", "> ");
    (void)scanf("%ld", &nws);
    (void)getchar();
    alambd = 10.;
/* prepare histogram */
    x0 = -.5;
    nx = 30;
    delx = 1.;
/* fill histogram with Poisson probabilities */
    i__1 = nx;
    for (i = 1; i <= i__1; ++i) {
	i__2 = i - 1;
	hist[i - 1] = sdpois_(&i__2, &alambd);
    }
/* prepare polyline */
    npl = 1000;
    dx = (double) nx * delx / (double) (npl - 1);
    sigma = sqrt(alambd);
/* compute points on polyline with normal probability density */
    i__1 = npl;
    for (i = 1; i <= i__1; ++i) {
	xpl[i - 1] = x0 + (double) (i - 1) * dx;
	ypl[i - 1] = sdnorm_(&xpl[i - 1], &alambd, &sigma);
    }
/* prepare texts and caption */
    (void)memset(tx, ' ', 75);
    (void)memset(ty, ' ', 75);
    (void)memset(capt, ' ', 75);
    (void)strncpy(tx, "k", 1);
    ltx = 1;
    (void)strncpy(ty, "P(k)", 4);
    lty = 4;
    (void)strncpy(capt, "Poisson (histogram) and Gaussian (cont. line)", 45);
    lcapt = 45;
/* produce graphics by call of GRHSCV */
    grhscv_(xpl, ypl, &npl, hist, &x0, &delx, &nx, tx, &ltx, ty, &lty,
           capt, &lcapt, &nws);
    return 0;
} /* main */

File: runme/mcgen/local/datan/exasrc/e6gr.f
      PROGRAM E6GR
      IMPLICIT DOUBLE PRECISION (A-H,O-Z)
      DIMENSION HIST(100),XPL(1000),YPL(1000)
      CHARACTER*75 TX,TY,CAPT
C identify program to user
      WRITE(*,*)' Program E6GR demonstrates use of GRHSCV'
      WRITE(*,*)' '
C ask for number of workstation
      CALL GRNBWS()
      WRITE(*,*)' Please, enter number of workstation:'
      WRITE(*,*)' >'
      READ(*,*)NWS
      ALAMBD=10.D0
C prepare histogram
      X0=-0.5D0
      NX=30
      DELX=1.D0
C fill histogram with Poisson probabilities
      DO 10 I=1,NX
        HIST(I)=SDPOIS(I-1,ALAMBD)
10    CONTINUE
C prepare polyline
      NPL=1000
      DX=DBLE(NX)*DELX/DBLE(NPL-1)
      SIGMA=SQRT(ALAMBD)
C compute points on polyline with normal probability density
      DO 20 I=1,NPL
        XPL(I)=X0+DBLE(I-1)*DX
        YPL(I)=SDNORM(XPL(I),ALAMBD,SIGMA)
20    CONTINUE
C prepare texts and caption
      TX='k'
      LTX=1
      TY='P(k)'
      LTY=4
      CAPT='Poisson (histogram) and Gaussian (cont. line)'
      LCAPT=45
C produce graphics by call of GRHSCV
      CALL GRHSCV(XPL,YPL,NPL,HIST,X0,DELX,NX,TX,LTX,
     +TY,LTY,CAPT,LCAPT,NWS)
      END
